### ISSUE TITLE
Add examples on how to modify fields (e.g. for Menderley's BibTeX field for files)

### ISSUE THREAD

**Issue by Zaeph** [Zaeph/org-roam-bibtex#4]
I was asked by @brianmcgillion on IRC to provide a working implementation to translate Menderley's BibTeX field for files.

Here's a prototype that works when your entire BibTeX file is generated via Menderley:
```el
(use-package org-roam-bibtex
  :after (org-ref)
  :custom
  (org-roam-bibtex-preformat-keywords
   '("=key=" "title" "url" "file" "author-or-editor"))
  (org-roam-bibtex-template
   '(("r" "ref" plain (function org-roam-capture--get-point) ""
      :file-name "${slug}"
      :head "#+TITLE: ${=key=}: ${title}\n#+ROAM_KEY: ${ref}

- tags ::

* ${title}
:PROPERTIES:
:Custom_ID: ${=key=}
:URL: ${url}
:AUTHOR: ${author-or-editor}
:NOTER_DOCUMENT: %(menderley-process-file-field \"${file}\")
:NOTER_PAGE:
:END:\n\n"
      :unnarrowed t)))
  :config
  (defun menderley-make-file-alist (str)
    "Return an alist cons cell with information on files.
The cons cell are of the form \(extension . path\)."
    (let (path)
      (with-temp-buffer
        (insert str)
        (goto-char (point-min))
        ;; Fix beginning
        (replace-regexp "^:\\(.*?\\)/" "/\\1/")
        ;; Fix ending
        (replace-regexp ":.*?$" "")
        (setq path (buffer-string))
        (cons (file-name-nondirectory path) path))))

  (defun menderley-process-file-field (string)
    "Process the ‘file’ BibTeX field as formatted by Menderley."
    (let* ((str-split (s-split ";" string))
           (candidates
            (mapcar #'menderley-make-file-alist
                    str-split)))
      (if (equal 1 (length candidates))
          (cdar candidates)
        (cdr (assoc (completing-read "File-type to use: " candidates)
                    candidates)))))

  (org-roam-bibtex-mode))
```

The key is in using `org-capture`'s sexp expansion via `%(…)`, which allows us to run functions taking as input BibTeX fields.  Then, the user only needs to write functions to process those strings.

It'd be nice if we could write an explanation of this behaviour in `README.md`, especially to highlight the order of expansion between BibTeX-expansion and `org-capture`-expansion.  We need to make it clear that people can *act* on those fields rather than just inserting them.

**Comment by myshevchuk** [comment]
It seems that `bibtex-completion` can handle Mendeley file field. https://github.com/tmalsburg/helm-bibtex#pdf-files

So this theoretically can be reduced to something like
```
(let ((entry (bibtex-completion-get-entry citekey))
  (bibtex-completion-apa-get-value "file" entry)

plus splitting the obtained string into individual filenames/cons-cells
```

**Comment by Zaeph** [comment]
> It seems that `bibtex-completion` can handle Mendeley file field. https://github.com/tmalsburg/helm-bibtex#pdf-files
> 
> So this theoretically can be reduced to something like
> 
> ```
> (let ((entry (bibtex-completion-get-entry citekey))
>   (bibtex-completion-apa-get-value "file" entry)
> 
> plus splitting the obtained string into individual filenames/cons-cells
> ```

Oh, thanks for looking it up!  Glad to know I've reinvented the wheel on that one, a not-so-round one at that.  I'll see whether we can add that to `README.md` in the coming days.

**Comment by brianmcgillion** [comment]
```
(defun process-file-field (key)
    (let ((entry (bibtex-completion-get-entry key)))
      (let* ((paths (bibtex-completion-find-pdf entry)))
        (if (equal 1 (length paths))
            (car paths)
          (completing-read "File to use: " paths)))))
```

**Comment by tmalsburg** [comment]
> It seems that bibtex-completion can handle Mendeley file field. https://github.com/tmalsburg/helm-bibtex#pdf-files

Bibtex-completion (now available as a separate MELPA package) also has a parser for Mendeley format. See function `bibtex-completion-find-pdf-in-field`.

**Comment by Zaeph** [comment]
> > It seems that bibtex-completion can handle Mendeley file field. https://github.com/tmalsburg/helm-bibtex#pdf-files
> 
> Bibtex-completion (now available as a separate MELPA package) also has a parser for Mendeley format. See function `bibtex-completion-find-pdf-in-field`.

Thanks, I should have looked it up before.  At least it reminded me how to use `completing-read`, so it wasn't all for nothing.

**Comment by Zaeph** [comment]
Addressed by #5.
